docs/dev/developer-guides/building-on-zksync/events.md

```diff
@@ -14,30 +14,49 @@ At zkSync, events behave the same way as in Ethereum.
 ## Events filtering
 
 Filtering is used to query indexed data and provide lower-cost data storage when the data is not required to be accessed on-chain.
+When filtering, you should load events by block ranges (0-1999, 2000-3999, ...) and index the result on your end. Otherwise you will get an error that says "block range should be less than or equal to 2000".
 
 These can be used in conjunction with the [Provider Events API](https://docs.ethers.io/v5/api/providers/provider/#Provider--event-methods) and with the [Contract Events API](https://docs.ethers.io/v5/api/contract/contract/#Contract--events).
 
+## Getting the events
 
+Here is an example to listen for smart contract events:
 
-```solidity
-    abi = [
-  "event Transfer(address indexed src, address indexed dst, uint val)"
-];
+```js
+import * as ethers from "ethers";
+const contractABI = require("./ABI_JSON")
 
-contract = new Contract(tokenAddress, abi, provider);
+const listenEvents = async () => {
+  const contractAddress = "<CONTRACT_ADDRESS>"
+  const provider = new ether.providers.WebSocketProvider(`wss://zksync2-testnet.zksync.dev/ws`)
 
-// List all token transfers *to* myAddress:
-contract.filters.Transfer(null, myAddress)
-//{
-//    "jsonrpc":"2.0",
-//   "id":1,
-//    "method":"eth_getLogs",
-//    "params":[{
-//       "fromBlock": "0x8e8e6",
-//        "toBlock": "0x8e8e6",
-//        "topics": [["0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"]]
-//    }]
-//}
+  const contract = new ethers.Contract(contractAddress, contractABI, provider);
+  
+  // starts listening to Transfer events on contract
+  contract.on("Transfer", (event) => {
+    // optional filter parameters
+    let options = {
+        filter: { INDEXED_PARAMETER: VALUE },
+        fromBlock: START_BLOCK_NUMBER, // e.g 15943000
+        toBlock: END_BLOCK_NUMBER, // e.g 15943100
+        data: event,
+    }
+    console.log(JSON.stringify(options, null, 4))
+  })
+}
+
+listenEvents()
 
 ```
-zkSync has a 10K log limit per response.
+
+
+- Provider: Your websocket provider through which you will retrieve the events data. Note that you need to use the websocket endpoint.
+- Contract address: The contract address whose events you want to track.
+- ABI: The ABI (Application Binary Interface) of the contract in JSON format.
+- Event name: The name of the event as defined in the smart contract. In this example we used the "Transfer" event from an ERC20 contract.
+- Indexed parameter (optional) : The indexed parameter of the event.
+- Block number (optional) The block number range for the events retrieval.
```

Comment:
Let's refer to start and end block number. And it is not really optional, after all we are stating in the beginning that if dev dones't use it, it will get an error. Would be better to explain this here